diff --git a/lib/tarballs/build.js b/lib/tarballs/build.js
index cbae61146324ef6ac0935c625b7858007ac238d1..33083bfccd7fad28aebec898c2e97fb9e1bd64be 100644
--- a/lib/tarballs/build.js
+++ b/lib/tarballs/build.js
@@ -69,9 +69,24 @@ const copyCoreYarnFiles = async (yarnRootPath, workspacePath) => {
 };
 async function build(c, options = {}) {
     (0, log_1.log)(`gathering workspace for ${c.config.bin} to ${c.workspace()}`);
-    await extractCLI(options.tarball ?? (await packCLI(c)), c);
-    await updatePJSON(c);
-    await addDependencies(c);
+    
+    if (true) { // Always using pnpm for us in this patch
+        try {
+            (0, log_1.log)('emptying', node_path_1.default.join(c.workspace()));
+            node_fs_1.rmSync(node_path_1.default.join(c.workspace()), {force: true, recursive: true});
+            // Use pnpm deploy to bundle the CLI into a single folder with production dependencies. 
+            await exec(`pnpm --filter . deploy --prod --legacy ${c.workspace()}`, { cwd: c.root });
+        } catch (e) {
+            (0, log_1.log)(e);
+            throw e;
+        }
+        await updatePJSON(c);
+    } else {        
+        await extractCLI(options.tarball ?? (await packCLI(c)), c);
+        await updatePJSON(c);
+        await addDependencies(c);
+    }
+   
     await (0, bin_1.writeBinScripts)({
         baseWorkspace: c.workspace(),
         config: c.config,
@@ -159,10 +174,11 @@ const addDependencies = async (c) => {
             }
         }
     }
-    else if ((0, node_fs_1.existsSync)(node_path_1.default.join(c.root, 'pnpm-lock.yaml'))) {
-        await (0, fs_extra_1.copy)(node_path_1.default.join(c.root, 'pnpm-lock.yaml'), node_path_1.default.join(c.workspace(), 'pnpm-lock.yaml'));
-        await exec('pnpm install --production', { cwd: c.workspace() });
-    }
+    // Not needed since we use pnpm deploy
+    // else if ((0, node_fs_1.existsSync)(node_path_1.default.join(c.root, '..' ,'pnpm-lock.yaml'))) {
+    //     await (0, fs_extra_1.copy)(node_path_1.default.join(c.root, '..', 'pnpm-lock.yaml'), node_path_1.default.join(c.workspace(), 'pnpm-lock.yaml'));
+    //     await exec('pnpm install --production', { cwd: c.workspace() });
+    // }
     else {
         const lockpath = (0, node_fs_1.existsSync)(node_path_1.default.join(c.root, 'package-lock.json'))
             ? node_path_1.default.join(c.root, 'package-lock.json')
